File: wiki-backend/src/main/java/com/internal/team/wiki/domain/doc/DocEntity.java
package com.internal.team.wiki.domain.doc;

import com.internal.team.wiki.global.BaseEntityTime;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import javax.persistence.Embedded;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Table(name ="docs")
@Entity
public class DocEntity extends BaseEntityTime {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String writerName;

    @Embedded
    private Title title;

    @Embedded
    private Contents contents;

    @Builder
    public DocEntity(final String writerName, final String title, final String contents) {
        this.writerName = writerName;
        this.title = new Title(title);
        this.contents = new Contents(contents);
    }

    public void change(final String title, final String contents) {
        this.title = new Title(title);
        this.contents = new Contents(contents);
    }
}

File: wiki-backend/src/main/java/com/internal/team/wiki/doc/domain/DocEntity.kt
package com.internal.team.wiki.doc.domain

import com.internal.team.wiki.global.BaseTimeEntity
import com.internal.team.wiki.user.domain.UserEntity
import javax.persistence.*

@Entity
@Table(name = "docs")
class DocEntity(

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    var user: UserEntity? = null,

    var writerName: String? = null,

    @Embedded
    var title: Title,

    @Embedded
    var contents: Contents,
) : BaseTimeEntity() {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null

    constructor(user: UserEntity, writerName: String, title: String, contents: String) : this(
        user = user,
        writerName = writerName,
        title = Title(title),
        contents = Contents(contents)
    )

    fun change(title: String, contents: String) {
        this.title = Title(title)
        this.contents = Contents(contents)
    }

    fun isUser(accessUserId: Long?): Boolean {
        return user?.id == accessUserId
    }
}